### PR TITLE
Add linter to govuk browser extension

### PR DESCRIPTION
**Pull request by kashifatcha**
## What

Add a linter to the govuk-browser-extension

## Why

To analyse the code for any potential errors or stylistic issues, making it easier to maintain code quality and consistency across the app.

https://trello.com/c/hgBWMrlB/2296-add-a-linter-to-govuk-browser-extension

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -4,7 +4,32 @@
     "jasmine-core": "^5.1.1"
   },
   "scripts": {
+    "lint:fix": "standardx --fix",
```
Comment (MartinJJones) [inline package.json:7]: After testing the `lint:fix` script locally, I think the command here needs to be updated to match `lint:js`, for example `"lint:fix": "standardx 'spec/javascripts/**/*.js' 'src/**/*.js' --fix"`

When running `lint:js` it will not include any linting errors for JS files in `spec/helpers`, but running `lint:fix` will look at all JS files including those stored in the helpers directory. I think this is why the linting fixes have also been applied to the `jasmine-query.js` file, which we likely do not want to modify.

**Review by MartinJJones — COMMENTED** [review]
Awesome work on adding this feature, will definitely help improve the Developer Experience when working on the govuk browser extension.

I've made a couple of small comments for things which may be worth a further look, just let me know if you need any further info or help on any of these 👍

File: package.json [hunk 2]
```diff
@@ -4,7 +4,32 @@
...
+  "eslintConfig": {
+    "env": {
+      "browser": true,
+      "jasmine": true
+    },
+    "rules":{
+      "no-var":0,
+      "no-unused-vars" : 0,
+      "no-use-before-define": 0
+    }
```
[the remark this answers is not in the document]
Comment (jon-kirwan) [reply package.json:24]: > Small non-blocking suggestion, the spacing for each rule is slightly different, we could keep it the same as other key value pairs in this file.

Yep, good suggestion @MartinJJones.

> It may have already been considered, but just in case, it appears we can also create a .eslintrc file in the root directory and move the eslintConfig rules here - https://www.npmjs.com/package/standardx#overview, for example:

I don't expect we'll need many more rules so I think it's probably okay to add the config rules in `package.json` for the time being which aligns with how other GOV.UK apps are setup e.g. [govuk_publishing_components](https://github.com/alphagov/govuk_publishing_components/blob/main/package.json#L21-L25) and [government-frontend](https://github.com/alphagov/government-frontend/blob/main/package.json#L28-L32)

File: package.json [hunk 3]
```diff
@@ -4,7 +4,32 @@
     "jasmine-core": "^5.1.1"
   },
   "scripts": {
+    "lint:fix": "standardx --fix",
+    "lint:js": "standardx 'spec/javascripts/**/*.js' 'src/**/*.js'",
```
[the remark this answers is not in the document]
Comment (jon-kirwan) [reply package.json:8]: It looks like the CSS files are rarely changed (4 years ago) so there's probably not much need to lint the CSS files for the time being.

File: package.json (hunk 2 above)
[the remark this answers is not in the document]
Comment (MartinJJones) [reply package.json:24]: Thanks @jon-kirwan, yeah probably don't need the separate `.eslintrc` file in that case, also good to keep it consistent with other GOV.UK apps, @kashifatcha think it's just the spacing issue to address here, I should have added that on a separate message to make it easier to resolve the conversation.

File: src/fetch-page-data.js [hunk 4]
```diff
@@ -12,33 +12,31 @@ chrome.runtime.sendMessage({
...
       allowedBuckets: allowedBuckets
     }
-  });
+  })
 
-  return buckets;
+  return buckets
 }
-
-undefined;
```
Comment (jon-kirwan) [inline src/fetch-page-data.js:44]: I think we should remove `undefined` from commit 4b2f6f1bd152082f72f8245a0c7ed4eaec81a257 (which will then remove the need to undo the change in commit 13db43a74d2444f17f97ef6c3a05d531c9e87aa8).

**Review by jon-kirwan — COMMENTED** [review]
Thanks Kash, well done on this 👍 - I've just asked for one small change below.

Also, usually we should exclude libraries like `jasmine-jquery` and polyfills from the linter. However, it's a slightly unusual here that it's included as a static asset rather than say pulled in from `node_modules` and, it's not likely to change again. So long as the extension works okay (which is does), i think it's fine to leave the changes in @MartinJJones - what are your thoughts?

File: src/fetch-page-data.js (hunk 4 above)
Comment (MartinJJones) [reply src/fetch-page-data.js:44]: Yeah, I agree, I also find the manual inclusion of the `jasmine-jquery` library here unusual. There is a similar approach in the publishing components gem, [modernizr.js](https://github.com/alphagov/govuk_publishing_components/blob/main/app/assets/javascripts/govuk_publishing_components/vendor/modernizr.js) for example, but these are stored in a `vendor` directory, which I think makes it clear that they are external libraries, as opposed to the directory of `helpers` used here, which could just be used for internal helpers we have created.

Would be good to review this again in the future to improve things, possibly by one or a mix of the approaches below:

- Move the libraries to a vendor directory
- Switch to using `node_modules`
- See if any of the libraries can be removed

**Comment by kashifatcha** [comment]
Changes have been made following suggestions. I have decided to revert the changes in jasmine-jquery.js. I have also decided not to add the .eslintrc file with the rules for the linter as this would be consistent with other apps across govuk.
